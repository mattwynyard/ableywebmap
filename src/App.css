#root {
  height: 100vh;
  overflow: hidden;
}

.login{
  position: absolute;
  right: 2%;
}

.brand {
  position: relative;
  left: 5%;
}

.map {
  position: absolute;
  top: 5%;
  padding: 0;
  margin: 0;
  height: 90%;
  width: 100%;
  z-index: 1;
  overflow: hidden;
}

.spinner {
  position: absolute;
  top: 50%;
  left: 50%;
  z-index: 1000;
}

.footer {
  position: absolute;
  top: 89%;
  height: 10%;
  width: 100%;
  font-size: 12px;  
}
