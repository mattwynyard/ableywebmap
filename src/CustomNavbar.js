import React from "react";
import {Navbar, Nav, Container, NavDropdown} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

function CustomNavbar() {
    return (
        <Navbar bg="light" expand="lg">
            <Navbar.Brand href="#home">
                <img
                src="logo192.png"
                width="30"
                height="30"
                className="brand"
                alt="logo"
            />   Property Viewer
            </Navbar.Brand>
            <Navbar.Toggle 
                aria-controls="basic-navbar-nav"    
            />
            <Navbar.Collapse 
                id="basic-navbar-nav"
            >
            <Nav className="me-auto"
            >
                <Nav.Link 
                    href="#home">Home
                </Nav.Link>
                <NavDropdown 
                    title="View" 
                    id="basic-nav-dropdown">
                    <NavDropdown.Item 
                        href="#action/3.1"
                        >Latest
                    </NavDropdown.Item>
                    <NavDropdown.Item 
                        href="#action/3.2"
                        >My Favorites
                    </NavDropdown.Item>
                    <NavDropdown.Item 
                    href="#action/3.3"
                        >Settings
                    </NavDropdown.Item>
                    <NavDropdown.Divider />
                    <NavDropdown.Item 
                        href="#action/3.4"
                        >Submit Viewing Request
                    </NavDropdown.Item>
                </NavDropdown>
                <Nav.Link className="login"
                    href="#home">Login
                </Nav.Link>
            </Nav>
            </Navbar.Collapse>
        </Navbar>
    )
}

export default CustomNavbar;