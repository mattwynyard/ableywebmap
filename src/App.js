import React, {useState} from "react";
import "./App.css";
import Mapper from "./Mapper.js";
import CustomFooter from "./CustomFooter";
import CustomNavbar from "./CustomNavbar";
import CustomSpinner from "./CustomSpinner"
function App() {

  const [status, setStatus] = useState("Loading");
  const [show, setShow] = useState(true);
  
  /**
   * Callback Hides spinner when map is loaded or error if not, called in mapper
   * @param {string} status - The Map load status
   */
  const setLoadStatus = (status) => {
    if (status === "loaded") {
      setShow(false);
    } else {
      setStatus(status);
    }
  }

  return (
    <React.Fragment>
      <CustomNavbar 
      />
      <Mapper 
        setLoadStatus={setLoadStatus}
      />
      <CustomSpinner 
        className="spinner" 
        show={show} 
        status={status}
      />
      <CustomFooter/>
    </React.Fragment>   
  );
}

export default App;
