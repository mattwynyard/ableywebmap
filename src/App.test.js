import React from 'react';
import ReactDOM from 'react-dom';
import CustomNavbar from './CustomNavbar';
import App from './App';
import { render, mount} from 'enzyme';

describe('App component', () => {
  test('it renders', () => {
    mount(<App />);
  });
 }) 