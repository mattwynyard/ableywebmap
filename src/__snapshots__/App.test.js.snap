// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<Navbar
  bg="light"
  className="navbar"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <NavbarBrand
    href="#home"
  >
    <img
      alt="logo"
      className="brand"
      height="30"
      src="logo192.png"
      width="30"
    />
       Property Viewer
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="me-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        disabled={false}
        href="#home"
      >
        Home
      </NavLink>
      <NavDropdown
        id="basic-nav-dropdown"
        title="View"
      >
        <DropdownItem
          href="#action/3.1"
        >
          Latest
        </DropdownItem>
        <DropdownItem
          href="#action/3.2"
        >
          My Favorites
        </DropdownItem>
        <DropdownItem
          href="#action/3.3"
        >
          Settings
        </DropdownItem>
        <DropdownDivider
          role="separator"
        />
        <DropdownItem
          href="#action/3.4"
        >
          Submit Viewing Request
        </DropdownItem>
      </NavDropdown>
      <NavLink
        className="login"
        disabled={false}
        href="#home"
      >
        Login
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
